api/v1alpha3/machine_types.go

```diff
@@ -245,6 +245,8 @@ type Bootstrap struct {
 // +kubebuilder:printcolumn:name="NodeName",type="string",JSONPath=".status.nodeRef.name",description="Node name associated with this machine",priority=1
 
 // Machine is the Schema for the machines API.
+//
+// Deprecated: This type will be removed in one of the next releases.
 type Machine struct {
```

Comment:
are you intentionally only marking the top level types as deprecated (not the individual spec/status/all types)?

Reply:
It was intentional, but the question is if it's the right thing to do.

Both marking the packages and types leads to linter findings. So not sure if we have to deprecate every type, constant, func, ....

Reply:
> So not sure if we have to deprecate every type, constant, func, ....

yeah that would be a lot, unless there is a way to auto-generate it somehow.

Reply:
I think (at least with staticcheck) flagging the package with deprecated has a similar effect.

As soon as I flagged the packages as deprecated I got a linter finding at every single import of those packages.